demo/midgard-sdk/src/linked-list.ts

```diff
@@ -31,6 +44,40 @@ export const CommonSchema = Data.Object({
 export type Common = Data.Static<typeof CommonSchema>;
 export const Common = CommonSchema as unknown as Common;
 
+export const KeyUnorderedLinkedListRedeemerSchema = Data.Enum([
+  Data.Literal("Init"),
+  Data.Literal("Deinit"),
+  Data.Literal("PrependUnsafe"),
+  Data.Literal("AppendUnsafe"),
+  Data.Literal("Remove"),
+]);
+
+export type KeyUnorderedLinkedListRedeemer = Data.Static<
+  typeof KeyUnorderedLinkedListRedeemerSchema
+>;
+export const KeyUnorderedLinkedListRedeemer =
+  KeyUnorderedLinkedListRedeemerSchema as unknown as KeyUnorderedLinkedListRedeemer;
+
+export const KeyOrderedLinkedListRedeemerSchema = Data.Enum([
+  Data.Literal("Init"),
+  Data.Literal("Deinit"),
+  Data.Literal("PrependSafe"),
+  Data.Literal("AppendSafe"),
+  Data.Literal("Insert"),
+  Data.Literal("Remove"),
+]);
+
+export type KeyOrderedLinkedListRedeemer = Data.Static<
+  typeof KeyOrderedLinkedListRedeemerSchema
+>;
+export const KeyOrderedLinkedListRedeemer =
+  KeyOrderedLinkedListRedeemerSchema as unknown as KeyOrderedLinkedListRedeemer;
+
+export const EmptyRootDataSchema = Data.Tuple([]);
+
+export type EmptyRootData = Data.Static<typeof EmptyRootDataSchema>;
+export const EmptyRootData = EmptyRootDataSchema as unknown as EmptyRootData;
+
```

Comment:
Remove these please. The redeemer is not abstracted in the current on-chain code.